docs/src/components/home/HeroEnd.tsx

```diff
@@ -1,98 +1,27 @@
 import * as React from 'react';
+import dynamic from 'next/dynamic';
+import { useInView } from 'react-intersection-observer';
```

Comment:
What do you think about moving this to the lab, so we can dog food https://github.com/mui-org/material-ui/issues/16663?

Reply:
Why, I don't get your point? Can you elaborate?

Reply:
This helper seems simple enough for us to implement a quick & dirty version and start experimenting with it before we expose something public.